File: Data Structures & Algorithms/contains-duplicate-ii/submission-0.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        hm = {}
        for index, num in enumerate(nums):
            if num in hm and index - hm[num] <= k:
                return True
            hm[num] = index

        return False

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
          unordered_map<int, int> hm;
  vector<int> result;
  for (auto num : nums) {
    hm[num]++;
  }

  vector<pair<int, int>> sorted(hm.begin(), hm.end());
  sort(sorted.begin(), sorted.end(),
       [](const auto &a, const auto &b) { return a.second > b.second; });

  for (auto [key, val] : sorted) {
    if (k > 0) {
      result.push_back(key);
      k--;
    }
  }
  return result;

    }
};
